// @AwsHelpers/index.js
const AWS = require( 'aws-sdk' )
const { SSM } = require( '@AWSHelpers/node_modules/aws-sdk' )

const ssm = new SSM( {
	apiVersion: '2014-11-06',
	region: process.env.AWS_REGION && process.env.AWS_REGION !== 'undefined' ? process.env.AWS_REGION : 'us-east-1'
} )

var sqs = new AWS.SQS( {
	apiVersion: '2012-11-05'
} )

async function getByName ( name ) {
	try {
		console.log( 'trying to getByName - name = ' + name )
		var params = {
			Name: `/routeme/${process.env.STAGE}/${name}`,
			WithDecryption: true
		}
		var request = await ssm.getParameter( params ).promise()
		return request.Parameter.Value
	} catch ( error ) {
		console.error( 'Failed to retrieve param' )
		console.error( error )
		throw error
	}
}

function sendSqsMessage(body) {
    return sqs.sendMessage({
        MessageBody: JSON.stringify({
            queueUrl: process.env.QueueURL,
            ...body
        }),
        QueueUrl: process.env.QueueURL,
    }).promise()
}

function test() {
    console.log('This is from lambda');
    console.log('stage => ', process?.env?.Stage);
    return 'Hello from Layer'
}

module.exports = {
    sendSqsMessage,
    test,
    getByName
}
